notebooks/test_driven.clj

```diff
@@ -0,0 +1,42 @@
+^:nextjournal.clerk/no-cache
+(ns test-driven
+  (:require [clojure.test :refer [deftest is testing] :as t]
+            [nextjournal.clerk :as clerk]))
+
+;; 🧪 testing within a notebook
+
+;; viewer that runs tests when it encounters a test-var
+(defn- test-var? [x]
+  (not (nil? (when-let [v (get x :nextjournal.clerk/var-from-def)]
+               (println v)
+               (:test (meta v))))))
+
+(defn with-test-out->str [func]
+  (let [s (new java.io.StringWriter)]
+    (binding [t/*test-out* s]
+      (func))
+    (str s)))
+
+(defn test-runner-viewer [v]
+  (with-test-out->str #(t/run-test-var v)))
+
+
+(clerk/set-viewers!
+  [{:pred         test-var?
+    :transform-fn #(-> % :nextjournal.clerk/var-from-def test-runner-viewer)
+    :render-fn    #(v/html [:span.syntax-string.inspected-value %])}])
+
+(deftest foo
+  (is (= 1 2))
+  (is (= 1 1)))
+
+;; if we want to display `testing` results we'll need to wrap them somehow like
+;; this:
+(with-test-out->str
+  #(testing
+     (is (= [1] [2]))))
```

Comment:
we could potentially explore the ability to inline `testing` expressions and have clerk see them and run them just like it runs `deftest`s here. Might be a bit ergonomic, but is also a bit messy because top-level `testing` will always print test failures on ns reloads